Detectors/TOF/workflowIO/src/TOFRawWriterSpec.cxx

```diff
@@ -71,7 +72,7 @@ void RawWriter::run(ProcessingContext& pc)
   encoder.alloc(cache);
 
   int nwindowperorbit = Geo::NWINDOW_IN_ORBIT;
-  int nwindowintimeframe = o2::raw::HBFUtils::Instance().getNOrbitsPerTF() * nwindowperorbit;
+  int nwindowintimeframe = GRPGeomHelper::instance().getGRPECS()->getNHBFPerTF() * nwindowperorbit;
```

Comment:
GRPGeomHelper is a singleton helper to load relevant CCDB objects (particularly, you need GRPECS), but I don't see where do you init it in the workflow. See e.g. 
https://github.com/AliceO2Group/AliceO2/blob/05d28f5b1551fa4527bd64d57b6873f49e558bcd/Detectors/AOD/src/AODProducerWorkflowSpec.cxx#L1369
https://github.com/AliceO2Group/AliceO2/blob/05d28f5b1551fa4527bd64d57b6873f49e558bcd/Detectors/AOD/src/AODProducerWorkflowSpec.cxx#L2277
https://github.com/AliceO2Group/AliceO2/blob/05d28f5b1551fa4527bd64d57b6873f49e558bcd/Detectors/AOD/src/AODProducerWorkflowSpec.cxx#L2314
https://github.com/AliceO2Group/AliceO2/blob/05d28f5b1551fa4527bd64d57b6873f49e558bcd/Detectors/AOD/src/AODProducerWorkflowSpec.cxx#L1369
https://github.com/AliceO2Group/AliceO2/blob/05d28f5b1551fa4527bd64d57b6873f49e558bcd/Detectors/AOD/src/AODProducerWorkflowSpec.cxx#L2494https://github.com/AliceO2Group/AliceO2/blob/05d28f5b1551fa4527bd64d57b6873f49e558bcd/Detectors/AOD/src/AODProducerWorkflowSpec.cxx#L2494
for its preparation methods.

Reply:
Hi @shahor02 ,
I'll do it. I open the PR to force me to implement the change but I am not ready yet. I move to WIP

Reply:
I refurbished a bit the code.
For the moment only TOF raw decoder uses the new feature.
From raw decoder it depends the digit creation and encoding to ctf (then sync reco).
Async reco depends on ctf (written in sync).
MC is using a default number for TOF (128). For this I need to discuss the strategy with @sawenzel  to be consistent with all the rest

Reply:
Yes, but actually in MC we are not really using it (but to encode raw data)

Reply:
Can this be merged?
For the MC: As far as I know the simulation should not use it, it is event based.
Digitization would use the paramter from HBFUtils, also digit2raw conversion.
The value is then stored in the GRP, i.e. reconstruction should then get the correct value from the GRP automatically.

Reply:
In theory yes, but I am not sure MC updates the generated grp with optionally imposed HBFUtils value. Even if it is, as far as I can see, the generated grp's are not always used, e.g. the sim_challenge and FST ignore them. Perhaps @sawenzel can comment?

Reply:
In O2DPG, in principle the value of orbits passed via HBFUtils should be consistent to the value of orbits passed via the GRP CCDB objects. For the moment a GRPECS is always (re)constructed at the beginning of the workflow from the command line arguments to O2DPG workflow, so consistency is for the moment guaranteed.

Reply:
@sawenzel but in which case the generated GRPs are used?